### PR TITLE
Adds a setup and cli check for the recommended memory limit

### LINKED ISSUE #8642: Show warning if memory limit is below 512 MB (on CLI and setup check)
### Steps to reproduce
1. run sudo -u nginx ./occ files:scan --all

### Expected behaviour
It should pass OK. Or warn if memory limits are too low.

### Actual behaviour
PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 37748736 bytes) in /var/www/html/lib/private/Lock/DBLockingProvider.php on line 77
PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 8388616 bytes) in /var/www/html/lib/private/Lock/DBLockingProvider.php on line 268

### Server configuration

**Operating system**:
CentOS

**PHP version:**
7.1 with default 128M memory limit

**Nextcloud version:** (see Nextcloud admin page)
13.0

### PR DESCRIPTION
**Pull request by weeman1337**
Introduces the `MemoryInfo` class that handles the memory limit query. I didn't put it into the `public` namespace.

Closes #8642

### CODE REVIEW

File: lib/private/Server.php [hunk 1]
```diff
@@ -273,6 +273,10 @@ public function __construct($webRoot, \OC\Config $config) {
 		});
 		$this->registerAlias('SystemTagManager', \OCP\SystemTag\ISystemTagManager::class);
 
+		$this->registerService(MemoryInfo::class, function() {
```
Comment (nickvergessen) [inline lib/private/Server.php]: this should not be required but work automatically

**Comment by rullzer** [comment]
Lets also try to get this into 14.

**Comment by skjnldsv** [comment]
Rebased and cleaned up.

File: lib/private/MemoryInfo.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<?php
```
Comment (blizzz) [inline lib/private/MemoryInfo.php]: declare strict types, please, and also the license header as found on other PHP files

Comment (weeman1337) [reply lib/private/MemoryInfo.php]: Thanks for the review. I added the license texts and PHP strict mode.

**Review by skjnldsv — APPROVED** [review]
Good for me! Awesome work!! :hugs:

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**